src/minefield/utils.tests.cpp

```diff
@@ -39,16 +38,43 @@ TEST(isInvalidPosition, should_return_false_if_mine_is_empty_or_with_mine)
     EXPECT_FALSE(Utils::isInvalidPosition(mineA.state));
     EXPECT_FALSE(Utils::isInvalidPosition(mineB.state));
 }
+TEST(nameExists, should_return_false_if_player_vector_is_empty) 
+{
+    EXPECT_FALSE(Utils::nameExists("PlayerName", Players{}));
+}
 
-//int countOpponentMines(Player const& player, Players const& players);
+TEST(nameExists, should_return_true_if_name_exists_in_player_vector)
+{
+    Players players = Players{};
+    Player p1;
 
-//TEST(countOpponentMines, should_return_3_mines)
-//{
-//    GameContext context;
-//    Player player1;
-//    Player player2;
-//    EnterMineFn enterMine;
-//    enterMine(context, player1);
-//}
+    p1.name = "PlayerName";
+    players.emplace_back(p1);
+    EXPECT_TRUE(Utils::nameExists("PlayerName", players));
+}
+
+TEST(checkBoardFull, should_return_true_if_board_is_full)
+{
+    unsigned int width = 5;
+    unsigned int height = 5;
+    Players players;
+    Board board;
+    Utils::initializeBoard(board, height, width);
+    
+    for (unsigned int x = 0; x < board.size(); x++)
+    {
+        for (unsigned int y = 0; y < board[0].size(); y++)
+        {
+            board[x][y].state = PositionState::WithMine;
```

Comment:
#nmi is this iteration being done conceptually correct? Shouldn't it be `board[y][x]`, plus inverting the for loops? (Making `y` the top-level one and `x` the nested one)